packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -461,17 +489,23 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 		this.isAck = false;
 	}
 
-	private async unpackMessage(data: ArrayBuffer | Buffer, isBinary: boolean): Promise<GatewayReceivePayload | null> {
-		const decompressable = new Uint8Array(data);
-
+	private async unpackMessage(data: Data, isBinary: boolean): Promise<GatewayReceivePayload | null> {
 		// Deal with no compression
 		if (!isBinary) {
-			return JSON.parse(this.textDecoder.decode(decompressable)) as GatewayReceivePayload;
+			try {
+				return JSON.parse(data as string) as GatewayReceivePayload;
+			} catch {
+				// Either bun emulates the rejection as a message event or discord sent us invalid JSON, either way, we can't do anything with it
```

Comment:
To clarify: Bun doesn't implement compression for client-side WebSocket yet. Only on the server

Reply:
That comment is related to the linked issue in the PR description (https://github.com/oven-sh/bun/issues/3392) 😄